src/DOM/Node/MutationObserver.purs

```diff
@@ -0,0 +1,47 @@
+module DOM.Node.MutationObserver
+  ( MutationObserver
+  , MutationObserverInitFields
+  , mutationObserver
+  , observe
+  , disconnect
+  , takeRecords
+  ) where
+
+import Prelude
+
+import Control.Monad.Eff (Eff, kind Effect)
+import DOM (DOM)
+import DOM.Node.MutationRecord (MutationRecord)
+import DOM.Node.Types (Node)
+
+foreign import data MutationObserver :: Type
+
+type MutationObserverInitFields =
+  ( childList :: Boolean
+  , attributes :: Boolean
+  , characterData :: Boolean
+  , subtree :: Boolean
+  , attributeOldValue :: Boolean
+  , characterDataOldValue :: Boolean
+  , attributeFilter :: Array String
+  )
+
+foreign import mutationObserver
+  :: forall eff
+   . (MutationRecord -> MutationObserver -> Eff (dom :: DOM | eff) Unit)
+  -> Eff (dom :: DOM | eff) MutationObserver
+
+foreign import _observe :: forall eff r. Node -> Record r -> MutationObserver -> Eff (dom :: DOM | eff) Unit
```

Comment:
This might throw "An invalid or illegal string was specified"
https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver#MutationObserverInit